website/cue/reference/components.cue

```diff
@@ -49,6 +49,15 @@ components: {
 
 		configuration: #Schema
 
+		// This field should be used in cases where auto-generating example configs using fields
+		// like `common` and `required` breaks down. If no `example_config` is set on a component,
+		// the example config is auto-generated; if `example_config` is set, then this map will be
+		// used to generate the example directly.
+		example_config?: {
+			common:    #Fields
+			advanced?: #Fields
+		}
```

Comment:
Each component already has an `examples` root level key, so we don't need this.

Ex: https://github.com/vectordotdev/vector/blob/43dd342c7ef02a18ddc53620426b394f7386c269/website/cue/reference/components/transforms/metric_to_log.cue#L65

Reply:
@binarylogic This is different. This is for the example configuration at the top of the config, for example [here](https://vector.dev/docs/reference/configuration/transforms/route/#example-configurations).